encrypted_json_fields/encryption.py

```diff
@@ -0,0 +1,305 @@
+                if prefix == FERNET_PREFIX:
+                    # Fernet token directly
+                    return crypter._is_encrypted_internal(without_prefix)
+                else:
+                    # AES: urlsafe_b64decode and check
+                    try:
+                        raw_encrypted = base64.urlsafe_b64decode(without_prefix)
+                        return crypter._is_encrypted_internal(raw_encrypted)
+                    except Exception:
+                        return False
+
+        # No prefix => legacy Fernet
+        return self._is_legacy_data(data)
+
+    def encrypt_str(self, s: str) -> str:
+        if self.encryption_disabled or self.is_encrypted(s.encode("utf-8")):
+            return s
+        encrypted = self.encrypt(s.encode("utf-8"))
+        return encrypted.decode("utf-8")
+
+    def decrypt_str(self, value: str) -> str:
+        if self.encryption_disabled:
+            return value
+
+        decrypted = self.decrypt(value.encode("utf-8"))
+        return decrypted.decode("utf-8")
+
+    def encrypt_values(self, data, json_skip_keys=None, encoder=None):
+        if self.encryption_disabled:
+            return data
+
+        if json_skip_keys is None:
+            json_skip_keys = []
+
+        if isinstance(data, (list, tuple, set)):
+            return [self.encrypt_values(x, json_skip_keys=json_skip_keys,
+                                        encoder=encoder) for x in data]
+
+        if isinstance(data, dict):
+            return {
+                key: value if key in json_skip_keys else self.encrypt_values(
+                    value, json_skip_keys=json_skip_keys, encoder=encoder)
+                for key, value in data.items()
+            }
+
+        if isinstance(data, (int, float, bool, str)):
+            try:
+                encoded_data = repr(data)
+                return self.encrypt_str(encoded_data)
+            except Exception as e:
+                raise ValueError(
+                    f"Failed to encrypt value: {data} (Error: {str(e)})")
+
+        try:
+            encoder_obj = encoder() if encoder else json.JSONEncoder()
+            encoded_data = encoder_obj.encode(data)
+            return self.encrypt_str(encoded_data)
+        except Exception as e:
+            raise ValueError(
+                f"Failed to encode and encrypt value: {data} (Error: {str(e)})")
+
+    def decrypt_values(self, data):
+        if self.encryption_disabled:
+            return data
+
+        if isinstance(data, (list, tuple, set)):
+            return [self.decrypt_values(x) for x in data]
+
+        if isinstance(data, dict):
+            return {key: self.decrypt_values(value) for key, value in data.items()}
+
+        if not isinstance(data, str):
+            return data
+
+        if not self.is_encrypted(data):
+            return data
+
+        decrypted_data = self.decrypt_str(data)
+
+        try:
+            return literal_eval(decrypted_data)
```

Comment:
Why do we need to eval this again? Anytime we introduce eval it poses security risk, so let's make sure this is the only path forward.

Reply:
It's mainly for handling converting strings into float, bool, none, or int. I'll see if I can do that without literal_eval